x/staking/keeper/msg_server.go

```diff
@@ -855,12 +855,18 @@ func (k msgServer) RedeemTokensForShares(goCtx context.Context, msg *types.MsgRe
 	// Similar to undelegations, if the account is attempting to tokenize the full delegation,
 	// but there's a precision error due to the decimal to int conversion, round up to the
 	// full decimal amount before modifying the delegation
-	shares := sdk.NewDecFromInt(shareToken.Amount)
+	shareDenomSupply := k.bankKeeper.GetSupply(ctx, shareToken.Denom)
+	shares := delegation.Shares.Mul(sdk.NewDecFromInt(shareToken.Amount)).QuoInt(shareDenomSupply.Amount)
```

Comment:
Apologies for the confusion, but can you elaborate on why this is inaccurate in the case of a validator having pending rewards or being slashed?

The LSM token is intended to map 1:1 with shares (it is minted based on shares, not tokens), so I was under the assumption that it would naturally capture rewards/slashing.

Reply:
> The LSM token is intended to map 1:1 with shares (it is minted based on shares, not tokens), so I was under the assumption that it would naturally capture rewards/slashing.

You are right. Sorry, I got it the wrong way - the simulation tests were failing due to these lines, which don't use a 1:1 mapping between LSM tokens to shares.